src/data-workspace/display/display.js

```diff
@@ -0,0 +1,183 @@
+import { useConfig } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { Button, IconLaunch16 } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { useExchangeContext } from '../../exchange-context/index.js'
+import { Warning } from '../../shared/index.js'
+import { Table } from '../table/index.js'
+import styles from './display.module.css'
+
+// TBD re: data conversion for formatting
+
+const convertToObjectFormat = ({
+    data,
+    dx_index,
+    ou_index,
+    pe_index,
+    value_index,
+}) => {
+    const objectFormat = {}
+
+    for (const row of data?.rows) {
+        if (
+            !Object.prototype.hasOwnProperty.call(objectFormat, row[ou_index])
```

Comment:
Why are you using `Object.prototype.hasOwnProperty.call(objectFormat, row[ou_index])`?
Wouldn't one of these do the same thing?
```js
objectFormat.hasOwnProperty(row[ou_index])
// OR
row[ou_index] in objectFormat
```

Reply:
This was because of our linter rules which give a `error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins` when you use hasOwnProperty directly on the object. I don't think the [justification](https://eslint.org/docs/latest/rules/no-prototype-builtins) eslint provides really makes sense in this context as we trust the content from the server and the object is not being created with `Object.create()`, so I could have disabled this rule, but I guess I didn't want to disable the lint rules if there was an alternative?

I have now changed it to use `in` with the implementation I've now pushed. Since we're making a map like `{objUid: {deUid: {peUid: value}}}`, the `in` check could lead to a problem if we have the uid `constructor` (valid DHIS2 uid and inherited property of objectFormat), and am okay with living with the infinitesimal chance of collision.

Reply:
Great analysis. I would be equally fine with either changing it back to the original implementation, or the `in` based logic you have now.